Sources/Sugar/Authentication/Updatable.swift

```diff
@@ -0,0 +1,23 @@
+import Vapor
+
+public protocol Updatable: class {
+    associatedtype Update
+    func preUpdate(on req: Request) -> Future<Void>
+    func update(_ update: Update) throws
+}
+
+extension Updatable {
+    public func preUpdate(on req: Request) -> Future<Void> {
+        return req.future()
+    }
+}
+
+extension Updatable where Update: Decodable {
+    public func applyUpdate(on req: Request) -> Future<Self> {
```

Comment:
I'm not sure, but should we align this with how the `Creatable` defines `create(..: Create)` and `create(..: Request)`?

Reply:
The naming was chosen because `update(on connectable: DatabaseConnectable)` is already defined in Fluent.